src/lxml/dtd.pxi

```diff
@@ -28,6 +28,11 @@ cdef class _DTDElementContentDecl:
     def __repr__(self):
         return "<%s.%s object name=%r type=%r occur=%r at 0x%x>" % (self.__class__.__module__, self.__class__.__name__, self.name, self.type, self.occur, id(self))
 
+    @property
+    def prefix(self):
+       _assertValidDTDNode(self, self._c_node)
+       return funicode(self._c_node.prefix) if self._c_node.prefix is not NULL else None
```

Comment:
There is a `funicodeOrNone()` function that does this NULL check dance for you.

Reply:
I would suggest replacing funicodeOrNone() as another commit, because there are multiple places in this file doing this check dance the same way.

Reply:
Works for me.